server/api/controllers/services/servicesController.js

```diff
@@ -18,6 +23,36 @@ function getServices(req, res, next) {
 }
 
 /**
+ * GET /services/{service}/asgs
+ */
+function getASGsByService(req, res, next) {
```

Comment:
Response to non-existent environment has status code 400 and error message. Response to non-existent service or slice is an empty array.
If no slice parameter is appended to the query string, blue and green slices are filtered out of the response, which is not intuitive. Suggest either mandatory slice parameter or unfiltered response if slice parameter is not specified.